deck/models.py

```diff
@@ -201,7 +199,12 @@ def get_rate(self):
     def rate(self, user, rate):
         rate_int = [r[0] for r in Vote.VOTE_RATES if rate in r][0]
         with transaction.atomic():
-            self.votes.create(user=user, rate=rate_int)
+            try:
+                vote = self.votes.get(user=user)
```

Comment:
Well, yes! But I'm going with `update_or_create`, that is discovered by now.

https://docs.djangoproject.com/en/1.7/ref/models/querysets/#update-or-create